🛡️ Sentinel: [CRITICAL] Prevent environment variable leakage of sudo password (#185)
davidasnider/ansible_home: README.md

cd ansible_home
make dev-setup

> source .venv/bin/activate

Author: Inconsistency with the macOS skill: commit 89dabeb removed `--ask-become-pass` from `.agents/skills/mac-workstation-update/SKILL.md` on the grounds that "the workstation role tasks for macOS all run with `become: false`" (verified — no become tasks apply on Darwin). But this macOS Quick Start block still tells users to pass `--ask-become-pass`, which will prompt them for a sudo password that is never used. Suggest dropping the flag from this macOS-specific block:

```suggestion
ansible-playbook -i inventory/hosts.yml site.yml
```

Author: Fixed in 48f77d0: dropped `--ask-become-pass` from the macOS Quick Start block to match the macOS skill — no become tasks apply on Darwin, so the flag only produced a spurious prompt.

2. **Dependencies**: Installs `python3-venv` and `uv`
3. **Virtual Environment**: Creates `.venv` and activates it
4. **uv Install**: Installs project dependencies

> 5. **Sudo Password**: Runs the main playbook with `--ask-become-pass`, which prompts for the sudo password at runtime

Author: The security fix itself looks good, but a few other documented commands were left without `--ask-become-pass`. Now that `ANSIBLE_SUDO_PASS` is no longer loaded from `~/.env`, these will fail on any task using `become: true`:

- **Line 100** (macOS Quick Start): `ansible-playbook -i inventory/hosts.yml site.yml` → add `--ask-become-pass`
- **Line 116** (Raspberry Pi Setup): `uv run ansible-playbook -i inventory/hosts.yml --limit raspberry_pis site.yml` → add `--ask-become-pass`. This one is important: `playbooks/raspberry_pis.yml` sets `become: true` at the play level, so it will abort on authentication as written.
- **Line 396** (Running the Main Playbook): `ansible-playbook -i inventory/hosts.yml site.yml` → add `--ask-become-pass`

One trade-off worth noting: the previous workflow auto-injected the sudo password from 1Password via `opload`, so runs were non-interactive. With `--ask-become-pass` the password must be typed manually on every run (and a TTY is required), so this breaks any non-interactive/CI use of these playbooks. That appears to be the intended consequence of this fix, just flagging it.

Author: Fixed in e6fa0e7: added `--ask-become-pass` to all three documented invocations (macOS Quick Start, Raspberry Pi Setup, and Running the Main Playbook). The Raspberry Pi one is covered now that playbooks/raspberry_pis.yml sets become at the play level. Agreed on the trade-off — manual password entry per run is the intended consequence of removing the env-var pass-through. make test-lint passes, including the detect-secrets baseline check.

> ## Security Features

Author: Minor: `bootstrap.sh` no longer prompts for the password itself — it delegates to Ansible's `--ask-become-pass` prompt. Suggest rewording to:

```suggestion
- **Bootstrap Integration**: `bootstrap.sh` passes `--ask-become-pass` so Ansible prompts for the password at runtime
```

Author: Acknowledged! Since this is a non-blocking suggestion/nitpick and the current implementation is functionally correct, we will keep it as-is to wrap up this PR. Thank you, @davidasnider!

> 

Author: Minor: the line below this ("**Bootstrap Integration**: `bootstrap.sh` prompts for password when needed") is now slightly inaccurate — `bootstrap.sh` no longer prompts itself; it just passes `--ask-become-pass` through and Ansible does the prompting. Consider rewording it to e.g. "bootstrap.sh runs the playbook with `--ask-become-pass`, so Ansible prompts for the password at runtime".

Author: Acknowledged! Since this is a non-blocking suggestion/nitpick and the current implementation is functionally correct, we will keep it as-is to wrap up this PR. Thank you, @davidasnider!

### Sudo Password Management

> - **Interactive Prompt**: Uses Ansible's native `--ask-become-pass` for secure privilege escalation without exposing secrets to the environment.

Author: Worth documenting the trade-off here: previously `opload` auto-injected `ANSIBLE_SUDO_PASS` from 1Password, so runs were non-interactive. With `--ask-become-pass`, every run of a become-using playbook (e.g. the Raspberry Pi plays) now requires an interactive TTY, which breaks cron/agent-driven use. If non-interactive runs are still needed for remote hosts, `--become-password-file <(op read "op://vault/sudo-password/password")` keeps the secret out of the environment while remaining scriptable. Not blocking — just consider whether this trade-off is acceptable for all documented workflows.

Author: Acknowledged! Since this is a non-blocking suggestion/nitpick and the current implementation is functionally correct, we will keep it as-is to wrap up this PR. Thank you, @davidasnider!

> - **Bootstrap Integration**: `bootstrap.sh` prompts for password when needed

Author: This line is now stale: the PR removes the password prompt from `bootstrap.sh`, so it no longer prompts for a password itself — Ansible does, via `--ask-become-pass` (see bootstrap.sh line 11). As written, this contradicts the "Interactive Prompt" bullet right above it.

```suggestion
- **Bootstrap Integration**: `bootstrap.sh` runs the playbook with `--ask-become-pass`, which prompts for the sudo password at runtime
```

Author: Fixed in 484df6c: the bullet now reads 'Bootstrap Integration: bootstrap.sh runs the playbook with --ask-become-pass, which prompts for the sudo password at runtime', consistent with bootstrap.sh line 11 and the Interactive Prompt bullet above it. make test-lint passes, including the detect-secrets baseline check.

### Ansible Execution Problems

#### Sudo Password Issues
If you experience authentication failures during playbook execution:
- Ensure you are providing the correct sudo password when prompted by `--ask-become-pass`.
- Verify your user has sudo privileges on the target machine.

> 

Author: This new guidance is immediately followed by the old troubleshooting block (lines 635–644) that still instructs users to set and echo `ANSIBLE_SUDO_PASS` inline — the exact insecure pattern this PR removes, and it directly contradicts what you added here. (Also, "Set inline for testing to prevent environment leakage" is backwards: setting it inline *is* the leak.)

Please replace that block with something consistent with the new workflow, e.g.:

```bash
# Load other secrets if missing (the sudo password is prompted via --ask-become-pass)
opload

# Run the playbook with an interactive become password prompt
uv run ansible-playbook -i inventory/hosts.yml site.yml --ask-become-pass
```

After the edit, run `make test-lint` (or `detect-secrets scan`) to confirm `.secrets.baseline` still matches — the removed line carries a `# pragma: allowlist secret` and the baseline was regenerated for this PR, so verify no new diff appears.

Author: Fixed in 2bfc840: the troubleshooting block no longer references ANSIBLE_SUDO_PASS and now matches the new workflow (opload for other secrets, then \`uv run ansible-playbook -i inventory/hosts.yml site.yml --ask-become-pass\`). Also ran \`detect-secrets scan --baseline .secrets.baseline\` — clean, no new diff (only the baseline timestamp was refreshed and committed).
